### DISCUSSION TITLE
Qual o nome desta técnica?

### DISCUSSION THREAD

**Discussion by igorthurow** [frontendbr/forum#1814] [Perguntas e Respostas]
Oi amigos, tudo bem?

Esta técnica utilizada na função **foo:** 

```
const baz = () => {
   const param1 = 1
   const param2 = 2

   const foo = () => bar(param1, param2)

   if (some) return foo()

   ...faz mais alguma coisa
   return foo()
}
```

É uma técnica parecida com aplicação parcial, mas escopa todos parâmetros para evitar a duplicidade da assinatura na invocação da função. Fiquei curioso se é uma técnica difundida e se tem um nome?

**Comment by tassoevan** [comment]
O conceito que identifica o que é feito em `foo` é _closure_, mas é a primeira vez que vejo esse tipo de aplicação.

**Comment by tassoevan** [comment]
Se bem que

> evitar a duplicidade da assinatura na invocação da função

parece o objetivo de _lazy evaluation_.

**Comment by wilsonneto-swe** [comment]
Me parece bastante como eu fazia os privates tempos atrás sem ES de ponta e sem TS rs Usava closures para deixar certos paramêtros ocultos dentro da func/objeto que eu retornava... Mas acho que para identificar a técnica em si precisaria ver o contexto maior dela...
